### ISSUE TITLE
[k8s] Change GPU base image to CUDA `devel` instead of `runtime` 

### ISSUE THREAD

**Issue by romilbhardwaj** [skypilot-org/skypilot#4122]
We currently use the cuda runtime image instead of devel image to reduce size (1.3GB vs 3.6GB). While this works for most workloads, some frameworks (such as deepspeed) require `nvcc`, which is [included only in devel image](https://github.com/NVIDIA/nvidia-docker/issues/1160).

**Suggested workaround**: manually specify base image with `--image-id nvidia/cuda:12.1.1-devel-ubuntu20.04`.

We should keep an eye out for users running into this issue, and consider switching to `devel` (at the cost of larger images, which take longer to provision).

**Comment by romilbhardwaj** [comment]
Deepspeed is broken on k8s if the user doesn't use a custom image. Need to fix this.

**Comment by romilbhardwaj** [comment]
Compared to our current -runtime image, the devel image takes 2.2x more time to pull (34s vs 76s on my network), and ~3.1x the disk space (7GB vs 2.25GB).

**Comment by romilbhardwaj** [comment]
Not stale - user ran into this today.

**Comment by chelsea0x3b** [comment]
Yeah I ran into this - to install flash attention I need nvcc, which the runtime doesn't have access to. Also wanted to suggested using a cudnn image which is basically just base image + cudnn.
